### PR TITLE
[Bug]: Tree, vertical spacing of branches and children is inconsistent

### PR DESCRIPTION
**Pull request by petdud**
This PR:
- Fixes incorrect level for subtrees
- Removes unused styles - grid subtree and flex
- Add padding-top for subtrees to address the bug by adding proper spacing 

## Previous Behavior

Missing spacing on top of sub trees

Screenshot 2023-07-31 at 11 44 29
 
## New Behavior

Adding padding-top on subtrees 

Screenshot 2023-07-31 at 11 44 16

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/08842860e5eef4755a2c0eee7b360d91b92593dc'>08842860e5eef4755a2c0eee7b360d91b92593dc</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/InfoButton-mount.html">9</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Avatar-mount_base.html">581</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Avatar-mount.html">598</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Button-mount_base.html">289</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Button-mount.html">287</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Field-mount_base.html">1050</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Field-mount.html">1081</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProvider-mount_base.html">652</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProvider-mount.html">639</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProviderWithTheme-mount_base.html">72</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProviderWithTheme-mount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">69</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">59</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">66</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/InfoButton-mount_base.html">13</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/InfoButton-mount.html">9</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/MakeStyles-mount_base.html">848</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/MakeStyles-mount.html">841</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Persona-mount_base.html">1664</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/Persona-mount.html">1653</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/SpinButton-mount_base.html">1370</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28681/perf-test-react-components/SpinButton-mount.html">1342</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-tree-0b6932ba-9e9f-4bc9-9d59-4d8152ff0cfe.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "bugfix: Tree, vertical spacing of branches and children is inconsistent",
+  "packageName": "@fluentui/react-tree",
+  "email": "petrduda@microsoft.com",
+  "dependentChangeType": "none"
+}
```
Comment (bsunderhus) [inline change/@fluentui-react-tree-0b6932ba-9e9f-4bc9-9d59-4d8152ff0cfe.json:7]:
```suggestion
{
  "type": "prerelease",
  "comment": "bugfix: Tree, vertical spacing of branches and children is inconsistent",
  "packageName": "@fluentui/react-tree",
  "email": "petrduda@microsoft.com",
  "dependentChangeType": "patch"
}
```

File: packages/react-components/react-tree/src/components/Tree/useTreeStyles.styles.ts [hunk 2]
```diff
@@ -17,15 +17,21 @@ const useStyles = makeStyles({
...
 export const useTreeStyles_unstable = (state: TreeState): TreeState => {
   const styles = useStyles();
   const isSubTree = state.level > 0;
+  const isNestedTree = state.level > 1;
+
   state.root.className = mergeClasses(
     treeClassNames.root,
     styles.root,
     isSubTree && styles.subtree,
+    isNestedTree && styles.nestedtree,
```
Comment (bsunderhus) [inline packages/react-components/react-tree/src/components/Tree/useTreeStyles.styles.ts:34]:
```suggestion
    ...shorthands.gridArea('subtree'),
    paddingTop: tokens.spacingVerticalXXS,
  },
});

export const useTreeStyles_unstable = (state: TreeState): TreeState => {
  const styles = useStyles();
  const isSubTree = state.level > 1;

  state.root.className = mergeClasses(
    treeClassNames.root,
    styles.root,
    isSubTree && styles.subtree,
```

seems like `state.level > 0` makes no sense, as it's always at least 1, so `isNestedTree` and `isSubtree` is the same thing

**Review by bsunderhus — APPROVED** [review]

**Pull request merged**